External Renderer/Assets/Scripts/Camera/CustomCamera.cs

```diff
@@ -1,183 +1,178 @@
-            if (!directory.Exists)
+            SaveCamera();
+            if (_camera == null)
             {
-                directory.Create();
+                enabled = false;
             }
-            folder = directory.FullName + Path.DirectorySeparatorChar;
         }
-        catch (System.Exception e)
+
+        /// <summary>
+        /// Create an internal reference to the Camera component attached to
+        /// this Gameobject. 
+        /// </summary>
+        public void SaveCamera()
         {
-            Debug.LogErrorFormat("Invalid Path \"{0}\". \n Threw {1}.",
-                folder, e.Message);
-            // Default to current Directory if provided one is invalid.
-            folder = Directory.GetCurrentDirectory();
-        }
+            _camera = GetComponent<Camera>();
 
-        return string.Format("{1}Screenshot-{0:yyyy-MM-dd_HH-mm-ss}.png",
-                System.DateTime.Now, folder);
-    }
+            if (_camera == null)
+            {
+                Debug.LogError("Could not find camera");
+            };
+        }
 
-    /// <summary>
-    /// Renders the current view of the Camera.
-    /// </summary>
-    /// <param name="folder">The folder in which to save the render.</param>
-    /// <param name="renderSize">The resolution of the rendered image.</param>
-    public void RenderImage(string folder = "", 
-        Vector2Int renderSize = default(Vector2Int))
-    {
-        // ensure screenshot size is at least 300x300 in size.
-        renderSize.Clamp(
-            new Vector2Int(300, 300), 
-            new Vector2Int(int.MaxValue, int.MaxValue));
+        /// <summary>
+        /// Write the image in <paramref name="render"/> to the render path.
+        /// </summary>
+        /// <param name="data">Bytes of the image to be saved.</param>
+        private void SaveRender(byte[] render)
+        {
+            FileManager file = new FileManager(_renderPath,
+                $"Render-{ DateTime.Now:yyyy-MM-dd-HH-mm-ss-fff-UTCzz}.png");
+            file.WriteToFile(render);
+            Debug.Log($"Saved render to { file.Path } at { DateTime.Now }.");
+        }
 
-        _camera.enabled = false;
-        RenderTexture renderTexture = new RenderTexture(renderSize.x, renderSize.y, 24);
-        _camera.targetTexture = renderTexture;
+        /// <summary>
+        /// Renders the current view of the Camera.
+        /// </summary>
+        /// <param name="renderSize">The resolution of the rendered image.</param>
+        public void RenderImage(Vector2Int renderSize = default)
+        {
+            // ensure screenshot size is at least 300x300 in size.
+            renderSize.Clamp(
+                new Vector2Int(300, 300),
+                new Vector2Int(int.MaxValue, int.MaxValue));
 
-        // Render the camera's view.
-        _camera.Render();
-        RenderTexture.active = renderTexture;
+            _camera.enabled = false;

```

Comment:
May need to change this to debug only, as in runtime, all cameras would be disabled (to not waste cpu or gpu resources automatically rendering).